### PR TITLE
enhance: Optimize import scheduling and add time cost metric (#36601)

### PR DESCRIPTION
**Pull request by bigsheeper**
1. Optimize import scheduling strategic:
a. Revise slot weights, calculating them based on the number of files and segments for both import and pre-import tasks.
b. Ensure that the DN executes tasks in ascending order of task ID.
2. Add time cost metric and log.

issue: https://github.com/milvus-io/milvus/issues/36600, https://github.com/milvus-io/milvus/issues/36518

pr: https://github.com/milvus-io/milvus/pull/36601

### CODE REVIEW

File: internal/datacoord/import_job.go [hunk 1]
```diff
@@ -91,15 +95,36 @@ type ImportJob interface {
...
 	GetFiles() []*internalpb.ImportFile
 	GetOptions() []*commonpb.KeyValuePair
+	Record(stage string, msg string, elapse bool)
 	Clone() ImportJob
 }
 
 type importJob struct {
 	*datapb.ImportJob
+
+	hasRecorded map[string]bool
```
Comment (czs007) [inline internal/datacoord/import_job.go:105]: add index state to task, and get rid of this hasRecorded.

Comment (bigsheeper) [reply internal/datacoord/import_job.go]: removed

**Comment by bigsheeper** [comment]
rerun ut

**Pull request merged**